de.fhdo.lemma.technology.technologydsl.extractor/src/de/fhdo/lemma/technology/technologydsl/extractor/TechnologyDslExtractor.xtend

```diff
@@ -0,0 +1,95 @@
+package de.fhdo.lemma.technology.technologydsl.extractor
+
+import de.fhdo.lemma.technology.DataFormat
+import de.fhdo.lemma.technology.Protocol
+import de.fhdo.lemma.technology.Technology
+import de.fhdo.lemma.technology.CommunicationType
+import de.fhdo.lemma.technology.JoinPointType
+import de.fhdo.lemma.technology.TechnologySpecificPrimitiveType
+import de.fhdo.lemma.technology.ServiceAspect
+
+/**
+ * Model-to-text extractor for the Technology DSL.
+ *
+ * @author <a href="mailto:jonas.sorgalla@fh-dortmund.de">Jonas Sorgalla</a>
+ */
+class TechnologyDslExtractor {
+
+    /**
+     * Extract TechnologyModel
+     */
+    def String extractToString(Technology techModel) {
+        val types = if (techModel.primitiveTypes !== null) {
```

Comment:
The `primitiveTypes` list should probably also be checked for emptyness. Maybe there is even a `nullOrEmpty` extension function in Xtend's standard library for `Iterable`s?